pkg/cache/internal/informers_map.go

```diff
@@ -162,9 +164,22 @@ func (ip *specificInformersMap) Get(gvk schema.GroupVersionKind, obj runtime.Obj
 	}
 
 	if started && !i.Informer.HasSynced() {
+		syncReturn := make(chan bool)
+		done := make(chan struct{})
+		go func() {
+			syncReturn <- cache.WaitForCacheSync(done, i.Informer.HasSynced)
```

Comment:
actually, why isn't think whole thing just 

```go
cache.WaitForCacheSync(ctx.Done(), i.Informer.HasSynced)
```

plus checking for the timeout?

We probably want to wire up `ip.stop` too via a `WithCancel` or somesuch.

Reply:
The first part of this comment seems totally reasonable to me. My gut says that the, "plus checking for the timeout?" is handled by leaving the select alone. Unfortunately though, the part about wiring up `ip.stop` I don't understand well enough to act on it.

Reply:
> plus checking for the timeout

I just meant checking the error from the context, IIRC

wrt to `ip.stop` -- we want to stop when one of the following things happens:

- the timeout fires from the context
- `ip.stop` is closed

the former we get for free from passing the context's done to WaitForCacheSync.  The latter I think we need something like

```go
// at the top
ctx, cancel := context.WithCancel(ctx)
defer cancel()

go func() {
  select {
    case <-ip.stop:
      cancel()
    case <- ctx.Done():
  }
}

// later
cache.WaitForSync(ctx.Done(), i.Informer.HasSynced)
```

Reply:
I believe that implementing the change that @joelanford suggested resolves this.